### PR TITLE
move from curl to reqwest

### LINKED ISSUE #60: Support platform native TLS implementation
Currently this depends on curl for fetching the pages, and the curl crate depends on openssl-sys. Switching to a more portable high level http library like https://crates.io/crates/reqwest should help with that, and is probably also a requirement for #53.

### PR DESCRIPTION
**Pull request by jcgruenhage**
Fixes #60

### CODE REVIEW

**Comment by jcgruenhage** [comment]
Apparently this breaks rust 1.28 support. How important is that?

**Comment by jcgruenhage** [comment]
Taking 1.31 once that is released sounds very good to me. `reqwest` isn't that old yet, yes, but really nice to use.

File: src/cache.rs [hunk 1]
```diff
@@ -61,19 +61,9 @@ impl Cache {
...
-            let mut transfer = handle.transfer();
-            try!(transfer.write_function(|data| {
-                buf.extend_from_slice(data);
-                Ok(data.len())
-            }));
-            try!(transfer.perform());
-        }
+        let mut resp = reqwest::get(&self.url)?;
+        let mut buf: Vec<u8> = vec![];
+        resp.copy_to(&mut buf)?;
```
Comment (dbrgn) [inline src/cache.rs]: The `unused_results` linter complains here. To avoid, use `let _bytes = resp.copy_to(&mut buf)?;`.

Comment (jcgruenhage) [reply src/cache.rs]: What do you think about logging it instead? It's not quite the same, but it is getting rid of the linter warning and shouldn't hurt anyway.

**Comment by jcgruenhage** [comment]
I don't understand the CI failure here.. The log line shouldn't change anything about the behavior when updating the cache, and the previous commit passed those tests.

File: src/cache.rs (hunk 1 above)
Comment (dbrgn) [reply src/cache.rs]: Ah, yeah, why not. Something like "Downloaded {} bytes".

**Comment by dbrgn** [comment]
@jcgruenhage since you run the command inside the `debug!()` macro, it is not executed when debug logs are not enabled :smiley: 

Solution: Assign the number of bytes to a variable and log that.

File: src/cache.rs (hunk 1 above)
Comment (jcgruenhage) [reply src/cache.rs]: I worded it slightly differently, "{} bytes downloaded", but the result is the same ^^

**Comment by jcgruenhage** [comment]
Oh, okay. Stupid me.

**Comment by jcgruenhage** [comment]
Interesting, github now shows force pushes. Neat!

**Comment by dbrgn** [comment]
> Interesting, github now shows force pushes. Neat!

Yeah, I noticed that too yesterday. Very nice.

Looks good now, thanks!

**Comment by dbrgn** [comment]
Merged in d9507007469c9be6c811ce820982c714bc2d2f26

**Pull request closed without merging**